ddtrace/tracer/payload.go

```diff
@@ -45,6 +45,10 @@ type payload struct {
 
 	// closed specifies the notification channel for each Close call.
 	closed chan struct{}
+
+	// confirm is a channel used to signal a payload has been sucessfully flushed.
+	// This is only used for testing purposes.
+	confirm chan<- struct{}
```

Comment:
This is a terrible practice that I've spent quite a bit of time removing. Let's not add it back. You may look around the alpha tracer I'm working on to see how I've solved it: https://github.com/DataDog/dd-trace-go/blob/v0.10.0/ddtrace/tracer/tracer_test.go#L980-L981

Reply:
Yes, please. There's a lot of testing hooks added into this repo, and it's not really a good programming practice IMO. Mostly my fault.